File: 2015/04/04.py
import _md5
def parser(target): 
    return target

def solver(target):
    def get_adventcoin(length):
        counter = 0
        while (counter := counter + 1) > 0:
            if _md5.md5(f"{target}{counter}".encode()).hexdigest().startswith('0' * length): 
                return counter
    
    yield get_adventcoin(5)
    yield get_adventcoin(6)

File: 2015/04/04.rs
aoc::main!();

fn preprocessing(input: &str) -> &str {
    input
}

fn part_1(target : &str) -> usize {
    get_adventcoin(target, 5)
}

fn part_2(target : &str) -> usize {
    get_adventcoin(target, 6)
}

fn get_adventcoin(target : &str, length: usize) -> usize {
    (1..)
    .find(|n| &format!("{:x}", md5::compute(format!("{}{}", target, n)))[..length] == "0".repeat(length))
    .unwrap()
}